utils/funcexp/funchelpers.h

```diff
@@ -393,21 +393,21 @@ inline bool calc_time_diff(int64_t time1, int64_t time2, int l_sign, long long*
              sec2 = 0,
              msec2 = 0;
 
-    year1 = (uint32_t)((time1 >> 48) & 0xffff);
-    month1 = (uint32_t)((time1 >> 44) & 0xf);
-    day1 = (uint32_t)((time1 >> 38) & 0x3f);
-    hour1 = (uint32_t)((time1 >> 32) & 0x3f);
-    min1 = (uint32_t)((time1 >> 26) & 0x3f);
-    sec1 = (uint32_t)((time1 >> 20) & 0x3f);
-    msec1 = (uint32_t)((time1 & 0xfffff));
-
-    year2 = (uint32_t)((time2 >> 48) & 0xffff);
-    month2 = (uint32_t)((time2 >> 44) & 0xf);
-    day2 = (uint32_t)((time2 >> 38) & 0x3f);
-    hour2 = (uint32_t)((time2 >> 32) & 0x3f);
-    min2 = (uint32_t)((time2 >> 26) & 0x3f);
-    sec2 = (uint32_t)((time2 >> 20) & 0x3f);
-    msec2 = (uint32_t)((time2 & 0xfffff));
+    year1 = isDateTime ? (uint32_t)((time1 >> 48) & 0xffff) : 0;
```

Comment:
I wonder if multiple isDataTime conditions produce lots of jmp instructions in ASM. If so I would better have one if-else.